base/test.jl

```diff
@@ -908,4 +909,16 @@ function detect_ambiguities(mods...; imported::Bool=false)
     collect(ambs)
 end
 
+"""
+The `GenericString` can be used to test generic string APIs that program to
+the `AbstractString` interface. In order to ensure that functions can work
+with string types besides the standard `String` type.
+"""
+immutable GenericString <: AbstractString
```

Comment:
It isn't mandatory for running all tests, or especially useful on its own, so Base.Test is an odd place to put it. This is "packages doing string-related things might want to optionally test against it." Granted the current TestHelpers is such a small stub, only used in a couple of places, and easy to miss.